cloudsmith/resource_service.go

```diff
@@ -272,6 +275,12 @@ func resourceService() *schema.Resource {
 				Computed:    true,
 				Sensitive:   true,
 			},
+			"warn_on_key_difference": {
+				Type:        schema.TypeBool,
+				Description: "Whether to warn if service's current key differs from key in state.",
+				Optional:    true,
+				Default:     true,
+			},
```

Comment:
Feels a bit weird to make this an "opt-in" which defaults to true. I'd be more inclined to make it an "opt-out" which defaults to false.

Reply:
I think either will be interchangeable, I'd personally be inclined to leave this as is, easy change anyway.